build-tools/packages/build-cli/src/lib/package.ts

```diff
@@ -667,6 +668,10 @@ export async function setPackageDependencies(
 				const dependencies = dev
 					? pkg.packageJson.devDependencies
 					: pkg.packageJson.dependencies;
+				if (dependencies === undefined) {
+					log?.warning(`Package has no dependencies or devDependencies: ${pkg.name}`);
```

Comment:
message makes it wound like there are both no deps and no dev deps, which is not the case. Also its inclear that this should be an warning (whats wrong with having no deps?).

Also warnings are not great as they tend to accumulate then bother people later when they wertent the one that caused them: I'd prefer errors or just working properly in most cases.